src/sentry/api/bases/organization_events.py

```diff
@@ -156,57 +149,33 @@ def get_snuba_query_args_legacy(self, request, organization):
                 'Boolean search operator OR and AND not allowed in this search.')
         return snuba_args
 
-    def next_event_id(self, *args):
+    def next_event_id(self, request, organization, snuba_args, event):
         """
         Returns the next event ID if there is a subsequent event matching the
-        conditions provided
+        conditions provided. Ignores the project_id.
         """
-        return self._get_next_or_prev_id(Direction.NEXT, *args)
+        next_event = eventstore.get_next_event_id(
+            event,
+            conditions=snuba_args['conditions'],
+            filter_keys=snuba_args['filter_keys']
+        )
 
-    def prev_event_id(self, *args):
+        if next_event:
+            return next_event[1]
+
+    def prev_event_id(self, request, organization, snuba_args, event):
         """
         Returns the previous event ID if there is a previous event matching the
-        conditions provided
+        conditions provided. Ignores the project_id.
         """
-        return self._get_next_or_prev_id(Direction.PREV, *args)
-
-    def _get_next_or_prev_id(self, direction, request, organization, snuba_args, event):
-        if (direction == Direction.NEXT):
-            time_condition = [
-                ['timestamp', '>=', event.timestamp],
-                [['timestamp', '>', event.timestamp], ['event_id', '>', event.event_id]]
-            ]
-            orderby = ['timestamp', 'event_id']
-            start = max(event.datetime, snuba_args['start'])
-            end = snuba_args['end']
```

Comment:
this is now capped to utcnow. Before this PR it came from snuba_args. 
Did you verify it is always capped to utcnow now or that it is not possible that an event timestamp may be later than utcnow ?

Reply:
Ouch, good point. It wasn't capped before, so somebody could've sent us an event with a timestamp (up to 60 seconds in future), and make a request with some future `end` value. With this change, they won't get any event in the future.

Getting a single event from Snuba (i.e. `EventManager.from_event_id`) is always capped to utcnow so maybe this makes sense.

Reply:
I added this for the project_event_details endpoint [here](https://github.com/getsentry/sentry/pull/14329). Since this code is not public currently (and might not ever become public in it's current format) I didn't add any metrics here.